src/libraries/System.Net.Http/src/System/Net/Http/DiagnosticsHandler.cs

```diff
@@ -116,6 +116,17 @@ private async ValueTask<HttpResponseMessage> SendAsyncCore(HttpRequestMessage re
             {
                 activity.Start();
 
+                // Set tags known at activity start
+                if (request.RequestUri is Uri requestUri && requestUri.IsAbsoluteUri)
+                {
+                    activity.SetTag("server.address", requestUri.Host);
+                    activity.SetTag("server.port", requestUri.Port);
+                    activity.SetTag("uri.full", DiagnosticsHelper.GetRedactedUriString(requestUri));
```

Comment:
OTel currently sets these attributes only if `activity.IsAllDataRequested`:

https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/4c69afa2fbb3b9d84b45bd73dcfad43331516e69/src/OpenTelemetry.Instrumentation.Http/Implementation/HttpHandlerDiagnosticListener.cs#L140-L143

Should the same logic be applied here too?

Reply:
Good catch, thanks! Fixed in https://github.com/dotnet/runtime/pull/104251/commits/7cf47e1429e1a9b6b336a0d05fafa8e9200072de.

Note that the OTel SDK [force-sets](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/4c69afa2fbb3b9d84b45bd73dcfad43331516e69/src/OpenTelemetry.Instrumentation.Http/Implementation/HttpHandlerDiagnosticListener.cs#L98-L105) `activity.IsAllDataRequested = false` in case it has not been created with an activity source.

@vishweshbankwar can you elaborate why is this needed?